packages/apollo-link-http/src/httpLink.ts

```diff
@@ -26,10 +26,16 @@ export default class HttpLink extends ApolloLink {
 
   public request(operation: Operation): Observable<FetchResult> | null {
     this.headers = (operation.context && operation.context.headers) || {};
+    const { operationName, variables, query, extensions } = operation;
     const request = {
-      ...operation,
-      query: print(operation.query),
-    };
+      operationName,
+      variables,
+      query: print(query),
+    } as any;
+
+    if (operation.context.includeExtensions && extensions) {
+      request.extensions = extensions;
```

Comment:
I was worried about confusing `extensions` and `context`. For now I think it is fine to separate the two into what is sent over the network vs kept in the chain. In the future, if context is sent to the server, the two may become redundant.